From dev-bounces@openvswitch.org Fri Apr 16 01:12:28 2010
Received: from FTLPMAILMX02.citrite.net (10.9.154.224) by
 LONPMAILMX02.citrite.net (10.30.224.163) with Microsoft SMTP Server (TLS)
 id 8.1.393.1; Fri, 16 Apr 2010 01:12:28 +0100
Received: from SMTP02.CITRIX.COM (10.9.154.239) by FTLPMAILMX02.citrite.net
 (10.9.154.224) with Microsoft SMTP Server id 8.1.393.1; Thu, 15 Apr 2010
 20:12:27 -0400
Received: from vps.nicira.com ([66.7.219.28])  by SMTP02.CITRIX.COM with
 ESMTP/TLS/DHE-RSA-AES256-SHA; 15 Apr 2010 20:12:26 -0400
Received: from localhost ([127.0.0.1]:52671 helo=vps.nicira.com)	by
 vps.nicira.com with esmtp (Exim 4.69)	(envelope-from
 <dev-bounces@openvswitch.org>)	id 1O2ZAB-0002yg-HU; Thu, 15 Apr 2010
 20:12:03 -0400
Received: from 173-164-153-21-sfba.hfc.comcastbusiness.net
 ([173.164.153.21]:40356 helo=leadville.nicira.com)	by vps.nicira.com with
 esmtps (TLSv1:AES256-SHA:256) (Exim 4.69)	(envelope-from <blp@nicira.com>)
 id 1O2ZA7-0002wZ-F5	for dev@openvswitch.org; Thu, 15 Apr 2010 20:11:59 -0400
Received: from hardrock.nicira.com ([172.16.0.20])	by leadville.nicira.com
 with esmtp (Exim 4.67)	(envelope-from <blp@nicira.com>)	id
 1O2ZA2-0007ew-8r; Thu, 15 Apr 2010 17:11:54 -0700
Received: from blp by hardrock.nicira.com with local (Exim 4.67)
 (envelope-from <blp@hardrock.nicira.com>)	id 1O2ZAD-0006DH-AS; Thu, 15 Apr
 2010 17:12:05 -0700
From: Ben Pfaff <blp@nicira.com>
To: "dev@openvswitch.org" <dev@openvswitch.org>
CC: Ben Pfaff <blp@nicira.com>
Sender: "dev-bounces@openvswitch.org" <dev-bounces@openvswitch.org>
Date: Fri, 16 Apr 2010 01:11:57 +0100
Subject: [ovs-dev] [PATCH 2/4] bridge: Minor cleanup in process_flow().
Thread-Topic: [ovs-dev] [PATCH 2/4] bridge: Minor cleanup in process_flow().
Thread-Index: Acrc+YAxgSk9VSKVRju55PrmbOELXg==
Message-ID: <1271376719-23848-2-git-send-email-blp@nicira.com>
References: <1271376719-23848-1-git-send-email-blp@nicira.com>
List-Help: <mailto:dev-request@openvswitch.org?subject=help>
List-Subscribe:
 <http://openvswitch.org/mailman/listinfo/dev_openvswitch.org>,
 <mailto:dev-request@openvswitch.org?subject=subscribe>
List-Unsubscribe:
 <http://openvswitch.org/mailman/options/dev_openvswitch.org>,
 <mailto:dev-request@openvswitch.org?subject=unsubscribe>
In-Reply-To: <1271376719-23848-1-git-send-email-blp@nicira.com>
Accept-Language: en-GB, en-US
Content-Language: en-US
X-MS-Exchange-Organization-AuthAs: Anonymous
X-MS-Exchange-Organization-AuthSource: FTLPMAILMX02.citrite.net
X-MS-Has-Attach: 
X-Auto-Response-Suppress: All
X-MS-TNEF-Correlator: 
x-sbrs: 5.3
x-mesageid: 92502208
x-ironport-server: ftlpip02.citrite.net
x-remote-ip: 66.7.219.28
x-policy: $ACCEPTED
x-ironport-anti-spam-filtered: true
x-ironport-anti-spam-result:
 AuoBAC9Ix0tCB9scmWdsb2JhbACPdot1FQEBAQEBCAsKBxEivlCFDgSDKw
x-ironport-av: E=Sophos;i="4.52,215,1270440000";   
 d="scan'208";a="92502208"
x-antiabuse: Sender Address Domain - openvswitch.org
x-source: 
x-source-args: 
x-source-dir: 
list-id: <dev_openvswitch.org.openvswitch.org>
errors-to: dev-bounces@openvswitch.org
list-post: <mailto:dev@openvswitch.org>
list-archive: <http://openvswitch.org/pipermail/dev_openvswitch.org>
x-beenthere: dev@openvswitch.org
x-mailman-version: 2.1.12.cp3
Content-Type: text/plain; charset="us-ascii"
MIME-Version: 1.0
X-Evolution-Source: imap://ianca@lonpmailmx01.citrite.net/
Content-Transfer-Encoding: 8bit

Should have no externally visible effect.
---
 vswitchd/bridge.c |    7 +++----
 1 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/vswitchd/bridge.c b/vswitchd/bridge.c
index baa889b..187115c 100644
--- a/vswitchd/bridge.c
+++ b/vswitchd/bridge.c
@@ -2255,16 +2255,13 @@ process_flow(struct bridge *br, const flow_t *flow,
         }
     }
 
-    /* MAC learning. */
-    out_port = FLOOD_PORT;
     /* Learn source MAC (but don't try to learn from revalidation). */
     if (packet) {
         update_learning_table(br, flow, vlan, in_port);
     }
 
     /* Determine output port. */
-    out_port_idx = mac_learning_lookup_tag(br->ml, flow->dl_dst, vlan,
-                                           tags);
+    out_port_idx = mac_learning_lookup_tag(br->ml, flow->dl_dst, vlan, tags);
     if (out_port_idx >= 0 && out_port_idx < br->n_ports) {
         out_port = br->ports[out_port_idx];
     } else if (!packet && !eth_addr_is_multicast(flow->dl_dst)) {
@@ -2274,6 +2271,8 @@ process_flow(struct bridge *br, const flow_t *flow,
          * on a bond and blackhole packets before the learning table is
          * updated to reflect the correct port. */
         return false;
+    } else {
+        out_port = FLOOD_PORT;
     }
 
     /* Don't send packets out their input ports. */
-- 
1.6.6.1


_______________________________________________
dev mailing list
dev@openvswitch.org
http://openvswitch.org/mailman/listinfo/dev_openvswitch.org

